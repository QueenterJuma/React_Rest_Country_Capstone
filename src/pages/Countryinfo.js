import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import { filterByCode } from '../redux/country/FetchCountry';
import { get } from '../redux/country/CountryInfoSlice';

const Countryinfo = () => {
  const { countryFiltered, error } = useSelector((state) => state.country);
  const dispatch = useDispatch();
  const { code } = useParams();
  useEffect(() => {
    if (code) {
      dispatch(filterByCode(code.toLowerCase()));
    }

    if (error) {
      dispatch(error);
    }
    return () => {
      dispatch(get());
    };
  }, [dispatch, code, error]);
  return (
    <div>
      <div>
        <BsArrowRightCircle />
      </div>
      {countryFiltered.length > 0 ? (
        <>
          <div className="info-container">
            <img
              src={countryFiltered[0].coatOfArms.png}
              alt={countryFiltered[0].coatOfArms.alt}
              className="info-image"
            />
            <p>{countryFiltered[0].name.common}</p>
          </div>
          <div className="country-data">
            <h1>
              <span>{countryFiltered[0].name.official}</span>
            </h1>
            <ul className="data">
              <li>
                <p>
                  Population:
                  {' '}
                  <span>
                    {countryFiltered[0].population}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Time zone:
                  {' '}
                  <span>
                    {countryFiltered[0].timezones}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Capital:
                  {' '}
                  <span>
                    {countryFiltered[0].capital}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Languages:
                  {' '}
                  <span>
                    {Object.values(countryFiltered[0].languages)
                      .map((item) => item)
                      .join(',')}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Currency:
                  {' '}
                  <span>
                    {countryFiltered[0].currency}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Subregion:
                  {' '}
                  <span>
                    {countryFiltered[0].subregion}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Latitute:
                  {' '}
                  <span>
                    {countryFiltered[0].latitute}
                    {' '}
                  </span>
                </p>
              </li>
              <li>
                <p>
                  Longitude:
                  {' '}
                  <span>
                    {countryFiltered[0].longitude}
                    {' '}
                  </span>
                </p>
              </li>
            </ul>
          </div>
        </>
      ) : (
        <div>No data found</div>
      )}
    </div>
  );
};

export default Countryinfo;
