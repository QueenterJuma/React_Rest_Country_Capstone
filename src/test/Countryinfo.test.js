import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, screen } from '@testing-library/react';
import Countryinfo from '../pages/Countryinfo';
import '@testing-library/jest-dom/extend-expect';

describe('Countryinfo', () => {
  const mockStore = configureStore();
  const initialState = {
    country: {
      countryFiltered: [
        {
          name: { common: 'United States' },
          population: 328239523,
          capital: 'Washington, D.C.',
          subregion: 'North America',
          languages: {
            eng: 'English',
          },
          timezones: [
            'UTC-12:00',
            'UTC-11:00',
            'UTC-10:00',
            'UTC-09:00',
            'UTC-08:00',
            'UTC-07:00',
            'UTC-06:00',
            'UTC-05:00',
            'UTC-04:00',
            'UTC+10:00',
            'UTC+12:00',
          ],
        },
      ],
    },
  };
  const store = mockStore(initialState);
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Countryinfo />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders country name correctly', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Countryinfo />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByText('United States')).toBeInTheDocument();
    expect(screen.getByText(328239523)).toBeInTheDocument();
    expect(screen.getByText('North America')).toBeInTheDocument();
  });
});
