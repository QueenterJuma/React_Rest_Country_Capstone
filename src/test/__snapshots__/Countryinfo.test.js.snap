// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countryinfo renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="back-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
        fillRule="evenodd"
      />
    </svg>
  </a>
  <div
    className="content1 spi"
  >
    <div
      className="box-img1"
    />
    <p
      className="banner-text"
    >
      United States
    </p>
  </div>
  <div
    className="country-data"
  >
    <ul>
      <li>
        <div
          className="val flex"
        >
          Population:
           
          <p>
            <span>
              328239523
            </span>
             
          </p>
        </div>
      </li>
      <li
        className="new-Color"
      >
        <div
          className="val flex"
        >
          Time zone:
           
          <p>
            <span>
              UTC-12:00
              UTC-11:00
              UTC-10:00
              UTC-09:00
              UTC-08:00
              UTC-07:00
              UTC-06:00
              UTC-05:00
              UTC-04:00
              UTC+10:00
              UTC+12:00
            </span>
             
          </p>
        </div>
      </li>
      <li>
        <div
          className="val flex"
        >
          Capital:
           
          <p>
            <span>
              Washington, D.C.
            </span>
          </p>
        </div>
      </li>
      <li
        className="new-Color"
      >
        <div
          className="val flex"
        >
          Languages:
           
          <p>
            English
          </p>
        </div>
      </li>
      <li>
        <div
          className="val flex"
        >
          Subregion:
           
          <p>
            <span>
              North America
            </span>
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
